src/encoded/types/cart.py

```diff
@@ -99,47 +88,46 @@ def get_or_create_cart_by_user(context, request):
     user = request.registry[COLLECTIONS]['user'].get(userid)
     if not user:
         raise HTTPBadRequest()
-    carts = _get_carts_by_user(request, userid, ['disabled', 'deleted '])
+    carts = get_cart_objects_by_user(request, userid, blocked_statuses)
+    return carts, user
+
+
+@view_config(context=Cart.Collection, request_method='GET', permission='save-carts', name='get-cart')
+def get_or_create_cart_by_user(context, request):
+    carts, user = _get_user_carts(request, ['disabled', 'deleted '])
     if not carts:
         cart = _create_cart(request, user)
+        cart_atids = None
```

Comment:
This None seems like it could be cleaned up a bit. One suggestion:
```python
def get_or_create_cart_by_user(context, request):
    carts, user = _get_user_carts(request, ['disabled', 'deleted '])
    carts = [cart['@id'] for cart in carts]
    if not carts:
        cart = _create_cart(request, user)
    request.response.status = 200
    return {
        'status': 'success',
        '@type': ['result'],
        '@graph': carts or [cart]
    }
```

Reply:
Changed, though I didn’t like going through a loop when I didn’t have to, that case would loop zero times so it’s probably not a penalty. Not my preferred choice though.